processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -496,7 +496,10 @@ static Supplier<Object> supplierFromObjectSelector(
       return () -> {
         final Object val = selector.getObject();
         if (val != null) {
-          NonnullPair<ExpressionType, Object[]> coerced = ExprEval.coerceListToArray((List) val, homogenizeMultiValue);
+          NonnullPair<ExpressionType, Object[]> coerced = ExprEval.coerceListToArray(
+              Arrays.asList((Object[]) val),
```

Comment:
this cast doesn't look right;
according to line 495 `clazz = selector.classOfObject()` supposed to be `List`; why cast it to `Object[]` ? 

if its already a `[]` then I'm not sure why there would be a need for `coerceListToArray` 

shouldn't it be true that `selector.getObject()`  returns a `selector.classOfObject()` ? I think if that's not true I would assume there is a bug somewhere else :)

Reply:
@kgyrtkirk thanks for the review. Earlier in the `PassThroughColumnValueSelector` we are setting the clazz as `List.class` for the type Array. So the `selector.getObject()` was returning Object Array though the clazz provided is List. 

I now updated the `PassThroughColumnValueSelector` to have `selector.classOfObject()` and  `selector.getObject()` in sync.